pandas/io/excel/_base.py

```diff
@@ -852,6 +853,14 @@ def __init__(self, path_or_buffer, engine=None):
                 ext = os.path.splitext(str(path_or_buffer))[-1]
                 if ext == ".ods":
                     engine = "odf"
+
+        if engine == "xlrd":
```

Comment:
The warning should not be issued when the parameter `engine="xlrd"` is passed explicitly.

Reply:
Hmm, if the engine is deprecated, I would expect that all uses should be discouraged. Explicit or implicit.

Reply:
xlrd is the only thing that will read legacy .xls files unfortunately, so I don't think we need to outright remove all usage of it but want the default to switch to openpyxl

Reply:
So are we already switching to openpyxl for everything other than .xls files (except of course .ods files and maybe .xlsb files)

Reply:
Yea xlsx and xlsm files (the former I would hope is what the vast majority of people read nowadays)

Reply:
I have now changed the default engine to openpyxl and added a check to use xlrd for .xls files
This required quite some changes to the tests and a work-around for a rounding error in openpyxl.
See https://foss.heptapod.net/openpyxl/openpyxl/-/issues/1493